woocommerce/woocommerce
# Add/34331 add attributes modal

### All Submissions:






### Changes proposed in this Pull Request:

Adds the functionality to add a product attribute to the product. This will open up a Modal for users to add products.
Note: The ability to create new attributes and attribute terms is not added yet, as it would would drastically increase the size of this PR.

Mostly addresses #34331  .

https://user-images.githubusercontent.com/2240960/195096611-c217b1da-d983-4f83-927e-7c8f311515f1.mp4

### How to test the changes in this Pull Request:

1. Load this branch, build it, and make sure you have the `new-product-management-experience` feature flag enabled (you can do so using the latest version of the Beta tester within the mono repo).
2. Go to **Products > Add new (MVP)** and fill out some of the initial details like name and price
3. Move down to the attributes section and select **Add first attribute** a Modal should popup that looks similar/same to the design: 5sAIeTRd9Yp7nSCT33BAWz-fi-5180%3A129070
4. A initial row should be added with an empty attribute and attribute term field, the attribute term field should be disabled and hitting the `remove` icon should not do anything.
5. Search for a category and select one, the attribute term field should be enabled now and you can search/select attribute terms (note that the attribute term field relies on the API for searching, it's not the smoothest at this point, but this will require separate updates to the SelectControl component, I do have a [tentative PR](https://github.com/woocommerce/woocommerce/pull/34949) in flight for this, but it will require some updating).
6. You should be able to click **+ Add another** which will add another row, you should be able to delete any of them except for the last one. So if you click **+ Add another** you can remove the first row if you want.
7. Once you have added items you can click **Add** this should add it to the product list, any empty rows should be ignored, this includes rows with a selected attribute, but no terms selected.

### Other information:




### FOR PR REVIEWER ONLY:
